src/main/kotlin/com/amazon/ionelement/impl/StructElementImpl.kt

```diff
@@ -61,8 +58,9 @@ internal class StructElementImpl(
                 fieldsByNameBackingField =
                     fields
                         .groupBy { it.name }
-                        .map { structFieldGroup -> structFieldGroup.key to structFieldGroup.value.mapToEmptyOrPersistentList { it.value } }
-                        .toMap().toPersistentMap()
+                        .map { structFieldGroup -> structFieldGroup.key to structFieldGroup.value.map { it.value }.toImmutableListUnsafe() }
+                        .toMap()
+                        .toImmutableMapUnsafe()
```

Comment:
🗺️ I wanted to create only a minimal set of extension functions, so I didn't create `Iterable<K, V>.toImmutableMap()`. Instead, I just use the equivalent `.toMap().toImmutableMapUnsafe()` here.